actionpack/test/controller/redirect_test.rb

```diff
@@ -119,6 +119,15 @@ def redirect_to_with_block_and_options
     redirect_to proc { { action: "hello_world" } }
   end
 
+  def redirect_to_out_of_scope_block
+    redirect_to self.class.class_eval(<<~END)
```

Comment:
Maybe stead this `class_eval` we could define this proc as a constant in Workshop?

Reply:
I originally wrote it as a constant (in the RedirectController class), but I thought it was easier to understand the test, especially with the `raise`, if the proc wasn't ~100 lines away.

I'll add that change, though.  I can see how the significance of the `class_eval` might not be immediately apparent.